msgs/yarp/thrift/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2018 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+set(LIBRARY_TARGET_NAME HumanStateMsg)
+set (HDE_PROJECT_NAME HumanDynamicsEstimation)
+
+
+yarp_idl_to_dir(HumanDynamicsEstimation/HumanState.thrift
+  ${CMAKE_CURRENT_BINARY_DIR}/autogenerated
+  SOURCES_VAR ${LIBRARY_TARGET_NAME}_GEN_SRC
+  HEADERS_VAR ${LIBRARY_TARGET_NAME}_GEN_HDR
+  INCLUDE_DIRS_VAR ${LIBRARY_TARGET_NAME}_INCLUDE_DIRS)
+
+include_directories(${LIBRARY_TARGET_NAME}_INCLUDE_DIRS)
+add_library(${LIBRARY_TARGET_NAME} SHARED ${${LIBRARY_TARGET_NAME}_GEN_SRC} ${${LIBRARY_TARGET_NAME}_GEN_HDR})
+target_link_libraries(${LIBRARY_TARGET_NAME} ${YARP_LIBRARIES})
```

Comment:
Are you sure you cannot limit the number of YARP components required to link against the resulting shared library?

Reply:
the list of yarp laibraries to be linked is `YARP::YARP_initYARP::YARP_OSYARP::YARP_sigYARP::YARP_devYARP::YARP_math`. I'm not sure if a subset of them is enough to be linked against the target. Let me know if you think, we can limit it.

Reply:
I'm not sure to be honest, you can inspect with `nm` or similar tools the symbols. Otherwise, just add `YARP_OS` and then proceed with linking. If you're on macOS, the linker will complain if symbols are missing (this project does not have the [usual tweak](https://github.com/robotology/blockfactory/blob/a05e6d0d192274ebe4f4fae57d8987c9dfa3cde0/CMakeLists.txt#L42-L50) I use to make it work also on linux, we should add it).

Reply:
acoording to the nm output such as `__ZN4yarp2os3idl10WireReader13readListBeginERNS1_9WireStateERj`, YARP::YARP_OS is fine. Commited the requested changes.